### ISSUE TITLE
Custom headers in StrawberryShake.Tools with '=' in value

### ISSUE THREAD

**Issue by rvanheest** [ChilliCream/hotchocolate#4410]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

In #4080 and #4384 custom headers were added for `StrawberryShake.Tools` such that we're now able to do something like `dotnet graphql init <url> <other arguments> --headers <key>=<value>`. In my use-case I have a particular way of authenticating with our services: we provide a header with `<key> = X-JWT-Assertion` and a base-64 encoded `<value>`. This `<value>` contains one or more `=` signs.

The current implementation of the `CustomHeaderHelper.ParseHeadersArgument()` is such that an argument (being `<key>=<value>`) is split on `=` and is only added to the list of custom headers when the splitted string contains 2 elements (so the argument contains at most 1 `=`).

In my use-case this means this header is rejected and not added to the list of custom headers. I'm not sure if there is a specific reason for this implementation, but wouldn't it be better to only split on the first occurrence of `=` in the argument string? Then everything before the first occurrence of `=` is the `<key>` and everything after that is the `<value>`.

### Steps to reproduce

`dotnet graphql init <url> <other arguments> --headers X-JWT-Assertion=mybase64=.encodedvalue=.` should yield a custom header `{ key = 'X-JWT-Assertion', value = 'mybase64=.encodedvalue=.' }`

### Product

Strawberry Shake

### Version

12.2.0-rc.2

**Comment by michaelstaib** [comment]
You want to try a fix?

**Comment by johan-lindqvist** [comment]
This was just a miss by me not realizing that there could be '=' characters in the custom header values. Nice fix!

**Comment by johan-lindqvist** [comment]
Would be nice to close this since there's a PR with a fix already done

**Comment by rvanheest** [comment]
@johan-lindqvist the PR hasn't been merged, indeed. However, as I'm looking through the changes in https://github.com/ChilliCream/hotchocolate/pull/4414/files, I noticed that one file (the actual change) was missing.
In my original commit (b5c14b6ac0da4cb6822a2841fa7083562671bbdb), besides making the actual change, I added a unit test for this bug. If you're looking at the "remaining change" in #4414 right now, the unit test is the only thing that is left to be merged. Looks like the actual change was made in a3de6a716e2099e19568f8f02034dc4a62feef95 on December 2nd, 2021.

In summary, if the unit test is not wanted, I think it is best to close the PR and this issue.
